### PR TITLE
Add detail to PostgreSQL doc

### PR DESCRIPTION
**Pull request by macraig**
Related to https://github.com/sourcegraph/sourcegraph/issues/16473

### CODE REVIEW

File: doc/admin/external_services/postgres.md [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 You can use your own PostgreSQL v12+ server with Sourcegraph if you wish. For example, you may prefer this if you already have existing backup infrastructure around your own PostgreSQL server, wish to use Amazon RDS, etc.
 
+## General recommendations
+
+If you choose to set up your own PostgreSQL server, please note **we do not recommend using the same PostgreSQL instance for different databases**. Although it is technically possible to point both databases at the same PostgreSQL instance, we suggest either:
```
Comment (efritz) [inline doc/admin/external_services/postgres.md:7]: It's technically possible _now_ but we also make no guarantee that some minor update may cause table names to conflict if they're pointed at the same instance.

It's a **requirement** for them to be different _databases_ and **strongly recommended** for them to be difference servers and/or hosts.

**Review by efritz — CHANGES_REQUESTED** [review]
See inline comment.

File: doc/admin/external_services/postgres.md (hunk 1 above)
Comment (macraig) [inline doc/admin/external_services/postgres.md:7]:
```suggestion
If you choose to set up your own PostgreSQL server, please note **we strongly recommended for each database to be set up in different servers and/or hosts**. We suggest either:
```

Comment (macraig) [reply doc/admin/external_services/postgres.md:7]: @efritz is that clearer? Feel free to edit

**Review by efritz — APPROVED** [review]

**Pull request merged**